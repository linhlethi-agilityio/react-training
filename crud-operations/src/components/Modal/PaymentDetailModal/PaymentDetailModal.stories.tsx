import type { Meta, StoryObj } from '@storybook/react';

// Components
import PaymentDetailModal from '.';

const meta: Meta<typeof PaymentDetailModal> = {
  component: PaymentDetailModal,
};

export default meta;
type Story = StoryObj<typeof PaymentDetailModal>;

export const Default: Story = {
  args: {
    isOpen: true,
    onClose: () => null,
    previewData: null,
  },
};
