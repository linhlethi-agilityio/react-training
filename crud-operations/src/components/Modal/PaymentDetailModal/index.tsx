import { memo } from 'react';
import { Box, Flex, Heading, Text } from '@chakra-ui/react';

// Components
import { CustomModal } from '@components';

// Types
import { Payment } from '@types';

interface PaymentDetailModalProps {
  isOpen: boolean;
  previewData: Payment | null;
  onClose: () => void;
}

const PaymentDetailModal = ({ isOpen, previewData, onClose }: PaymentDetailModalProps) => {
  const { name, paymentSchedule, billNumber, amountPaid, balanceAmount, date } = previewData || {};

  const PaymentDetailArray = [
    {
      name: 'Name',
      value: name,
    },
    {
      name: 'Payment Schedule',
      value: paymentSchedule,
    },
    {
      name: 'Bill Number',
      value: `INR ${billNumber}`,
    },
    {
      name: 'Amount Paid',
      value: `INR ${amountPaid}`,
    },
    {
      name: 'Balance amount',
      value: `INR ${balanceAmount}`,
    },
    {
      name: 'Date',
      value: date,
    },
  ];

  return (
    <CustomModal isOpen={isOpen} onClose={onClose} title="Payment Detail">
      <Box paddingX={10} paddingBottom={10}>
        {PaymentDetailArray.map((item) => {
          const { name, value } = item;

          return (
            <Flex justifyContent="space-between" marginTop={2.5} alignItems="center" key={`payment-item-${name}`}>
              <Heading size="xs">{name}:</Heading>
              <Text size="small">{value}</Text>
            </Flex>
          );
        })}
      </Box>
    </CustomModal>
  );
};

export default memo(PaymentDetailModal);
