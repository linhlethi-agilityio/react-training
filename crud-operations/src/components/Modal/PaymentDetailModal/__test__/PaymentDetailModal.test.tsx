import { render } from '@test-utils';
import '@testing-library/jest-dom';

// Components
import PaymentDetailModal from '..';

const mockProps = {
  isOpen: true,
  onClose: jest.fn(),
  previewData: null,
};

describe('PaymentDetailModal component', () => {
  test('should match snapshot for PaymentDetailModal', () => {
    const { container } = render(<PaymentDetailModal {...mockProps} />);

    expect(container).toMatchSnapshot();
  });
});
