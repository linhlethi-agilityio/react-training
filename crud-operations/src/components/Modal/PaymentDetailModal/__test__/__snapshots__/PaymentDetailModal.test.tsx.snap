// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentDetailModal component should match snapshot for PaymentDetailModal 1`] = `
<div
  aria-hidden="true"
  data-aria-hidden="true"
/>
`;
